tests/sbiutils_test.py

```diff
@@ -406,30 +407,113 @@ def test_kde(bandwidth, transform, sample_weights):
 
 
 @pytest.mark.parametrize(
-    "z_x", [True, False, None, "none", "independent", "structured"]
+    "z_x",
+    [
+        True,
+        False,
+        None,
+        "none",
+        "independent",
+        "structured",
+        "transform_to_unconstrained",
+        "invalid_value",
+    ],
+)
+@pytest.mark.parametrize(
+    "z_theta",
+    [
+        True,
+        False,
+        None,
+        "none",
+        "independent",
+        "structured",
+        "transform_to_unconstrained",
+        "invalid_value",
+    ],
+)
+def test_z_score_parser(z_x, z_theta):
+    if z_theta == "invalid_value":
+        with pytest.raises(ValueError, match="Invalid z-scoring option."):
+            z_score_parser(z_theta)
+
+    if z_x == "invalid_value":
+        with pytest.raises(ValueError, match="Invalid z-scoring option."):
+            z_score_parser(z_x)
+
+    if z_x != "invalid_value" and z_theta != "invalid_value":
+        result_x = z_score_parser(z_x)
+        result_theta = z_score_parser(z_theta)
+
+        assert result_x is not None, f"z_score_parser({z_x}) returned None"
+        assert result_theta is not None, f"z_score_parser({z_theta}) returned None"
+
+
+@pytest.mark.parametrize(
+    "z_x",
+    [
+        True,
+        False,
+        None,
+        "none",
+        "independent",
+        "structured",
+        "transform_to_unconstrained",
+    ],
 )
 @pytest.mark.parametrize(
-    "z_theta", [True, False, None, "none", "independent", "structured"]
+    "z_theta",
+    [
+        True,
+        False,
+        None,
+        "none",
+        "independent",
+        "structured",
+        "transform_to_unconstrained",
+    ],
 )
 @pytest.mark.parametrize("builder", [likelihood_nn, posterior_nn, classifier_nn])
 def test_z_scoring_structured(z_x, z_theta, builder):
     """
     Test that z-scoring string args don't break API.
     """
     # Generate some signals for test.
-    t = torch.arange(0, 1, 0.001)
+    t = torch.arange(0, 1, 0.1)
     x_sin = torch.sin(t * 2 * torch.pi * 5)
     t_batch = torch.stack([(x_sin * (i + 1)) + (i * 2) for i in range(10)])
 
+    num_dim = t_batch.shape[1]
+    x_dist = BoxUniform(low=-2 * torch.ones(num_dim), high=2 * torch.ones(num_dim))
+
     # API tests
+    # TODO: Test breaks at "mnle"
     if builder in [likelihood_nn, posterior_nn]:
-        for model in ["mdn", "made", "maf", "nsf"]:
+        for model in [
```

Comment:
this will let the number of effective test cases for this test grow exponentially, can you please double check the runtime of this test?